pkg/util/dynamichelper/dynamichelper_test.go

```diff
@@ -4,17 +4,92 @@ package dynamichelper
 // Licensed under the Apache License 2.0.
 
 import (
+	"context"
+	"errors"
 	"reflect"
 	"testing"
 	"time"
 
+	"github.com/golang/mock/gomock"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 
 	"github.com/Azure/ARO-RP/pkg/util/cmp"
+	mock_dynamichelper "github.com/Azure/ARO-RP/pkg/util/mocks/dynamichelper"
 )
 
+func TestEsureDeleted(t *testing.T) {
+	ctx := context.Background()
+
+	for _, tt := range []struct {
+		uname     string
+		grpkind   string
+		namespace string
+		name      string
+		wantErr   error
+		grpvr     *schema.GroupVersionResource
+	}{
+		{
+			uname:     "Test EnsureDeleted 1",
```

Comment:
1. Need more meaningfull test case names
2. What is the difference between this test case and the next one? They only seem to delete different objects of the same kind.